### PR TITLE
Add backend troubleshooting

### PR DESCRIPTION
**Pull request by elkoniu**
Migration of old documentation related to backend troubleshoting:
https://sssd.io/docs/users/troubleshooting/how-to-troubleshoot-backend.html

### CODE REVIEW

File: src/troubleshooting/backend.rst [hunk 1]
```diff
@@ -1,12 +1,626 @@
 Troubleshooting Backend
 =======================
 
-Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec ac ligula et felis aliquam posuere. Etiam ac augue vel tortor porttitor placerat vitae ac libero. Nunc a efficitur lectus. Aliquam malesuada congue arcu quis luctus. Quisque pretium dolor orci, ut ullamcorper felis tincidunt eget. Duis auctor felis ut maximus facilisis. Nam eu feugiat tortor.
+A ``backend``, often also called ``data provider``, is an SSSD child process that
+manages and ``creates the cache``. This process talks to LDAP server, performs
```
Comment (justin-stephenson) [inline src/troubleshooting/backend.rst:5]: a cache of what?

File: src/troubleshooting/backend.rst [hunk 2]
```diff
@@ -1,12 +1,626 @@
...
+Backend process and its files
+*****************************
 
-Nulla facilisi. Morbi venenatis tincidunt elit et efficitur. Praesent elementum ullamcorper sodales. Sed id justo id nisl tincidunt rhoncus. Nullam eleifend sit amet erat nec rhoncus. Proin id congue mi. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Mauris at arcu et nunc fermentum venenatis. Mauris ac efficitur purus, malesuada tristique nisi. Phasellus et fermentum dui, at dapibus ante. Duis efficitur iaculis arcu id accumsan. Sed sapien urna, placerat ut suscipit id, dignissim eget urna. Praesent eget risus fermentum, euismod nulla ut, maximus urna. Duis placerat risus at lorem accumsan fermentum. Nullam ultricies hendrerit euismod.
+There is one back-end process ``/usr/libexec/sssd/sssd_be`` for each
+domain configured within ``sssd.conf`` and ``enabled``. If domain is disabled
```
Comment (justin-stephenson) [inline src/troubleshooting/backend.rst:18]: s/If domain/If the domain/

File: src/troubleshooting/backend.rst [hunk 3]
```diff
@@ -1,12 +1,626 @@
...
+(by default) dedicated backend process will not be created.
```
Comment (justin-stephenson) [inline src/troubleshooting/backend.rst:19]: s/dedicated backend/then a dedicated backend/

File: src/troubleshooting/backend.rst [hunk 4]
```diff
@@ -1,12 +1,626 @@
...
+    # ls -l /var/lib/sss/db/
+    -rw-------. 1 root root 1286144 Mar 20 10:50 cache_IPA.PB.ldb
+    -rw-------. 1 root root 1286144 Mar 20 10:50 cache_AD.PB.ldb
+
+Data provider work flow
+***********************
+
+Backend provides several services: ``id``, ``auth``, ``access``, etc. (refer to sssd manual
+page for the full list of services). Each service is associated with one data
+provider through a configuration option, for example the id service is set to
```
Comment (justin-stephenson) [inline src/troubleshooting/backend.rst:76]: s/id service/identity service/

File: src/troubleshooting/backend.rst [hunk 5]
```diff
@@ -1,12 +1,626 @@
...
+
+.. code-block:: sssd-log
+
+    (Tue Nov 20 12:23:21 2020) [sssd[nss]] [cache_req_send] (0x0400): CR #59: New request 'User by name'
+    (Tue Nov 20 12:18:56 2020) [sssd[be[ldap.vm]]] [be_resolve_server_process] (0x1000): Saving the first resolved server
+    (Tue Nov 20 12:18:56 2020) [sssd[be[ldap.vm]]] [be_resolve_server_process] (0x0200): Found address for server master.ldap.vm: [192.168.100.20] TTL 0
+    (Tue Nov 20 12:18:56 2020) [sssd[be[ldap.vm]]] [sdap_uri_callback] (0x0400): Constructed uri 'ldap://master.ldap.vm:389'
+
+In the following chapters we will skip first part of the log line which is timestamp
+for the more clear log snippets.
```
Comment (justin-stephenson) [inline src/troubleshooting/backend.rst:181]: `In the following chapters we will exclude the first column timestamp in the log snippets` sounds a bit more clear IMO

**Comment by justin-stephenson** [comment]
Mostly minor changes requested.

File: src/troubleshooting/backend.rst (hunk 1 above)
Comment (elkoniu) [reply src/troubleshooting/backend.rst:5]: It was copy paste from original documentation. I trimmed it a little so it should be more straight forward now.

File: src/troubleshooting/backend.rst (hunk 2 above)
Comment (elkoniu) [reply src/troubleshooting/backend.rst:18]: Fixed

File: src/troubleshooting/backend.rst (hunk 3 above)
Comment (elkoniu) [reply src/troubleshooting/backend.rst:19]: Fixed

File: src/troubleshooting/backend.rst [hunk 6]
```diff
@@ -1,12 +1,626 @@
...
+    -rw-------. 1 root root 73870 Mar 20 10:50 sssd_AD.PB.log
+
+    # ls -l /var/lib/sss/db/
+    -rw-------. 1 root root 1286144 Mar 20 10:50 cache_IPA.PB.ldb
+    -rw-------. 1 root root 1286144 Mar 20 10:50 cache_AD.PB.ldb
+
+Data provider work flow
+***********************
+
+Backend provides several services: ``id``, ``auth``, ``access``, etc. (refer to sssd manual
```
[the remark this answers is not in the document]
Comment (elkoniu) [reply src/troubleshooting/backend.rst:74]: Fixed

File: src/troubleshooting/backend.rst (hunk 4 above)
Comment (elkoniu) [reply src/troubleshooting/backend.rst:76]: Fixed

File: src/troubleshooting/backend.rst (hunk 5 above)
Comment (elkoniu) [reply src/troubleshooting/backend.rst:181]: Fixed

**Comment by justin-stephenson** [comment]
One more suggested change.

File: src/troubleshooting/backend.rst [hunk 7]
```diff
@@ -1,12 +1,626 @@
...
+There is one back-end process ``/usr/libexec/sssd/sssd_be`` for each
+domain configured within ``sssd.conf`` and ``enabled``. If the domain is disabled
+(by default) then a dedicated backend process will not be created.
+Each backend process manages one SSSD domain defined in the configuration
+file and also its trusted domains (referred to as subdomains) that are
+auto-discovered at start up.
+
+Let’s take a look at this example configuration.
+
+.. code-block:: text
```
Comment (pbrezina) [inline src/troubleshooting/backend.rst:26]: `.. code-block:: ini`

File: src/troubleshooting/backend.rst [hunk 8]
```diff
@@ -1,12 +1,626 @@
...
+We have three domains configured, but only two enabled: ``IPA.PB`` and ``AD.PB``.
+Each enabled domain has its own cache file, which is shared between the parent domain
+and its subdomains and its own log file. The cache file is stored in
+``/var/lib/sss/db/$domain.ldb`` and the log file is located at
+``/var/log/sssd/$domain.log``.
+
+The following example shows us the SSSD backend processes and cache and log
+files location:
+
+.. code-block:: bash
```
Comment (pbrezina) [inline src/troubleshooting/backend.rst:57]: console

File: src/troubleshooting/backend.rst [hunk 9]
```diff
@@ -1,12 +1,626 @@
...
+                                  0xB7F0   0b10110111 11110000
+                                  0xF7F0   0b11110111 11110000
+                                  0x3FF0   0b00111111 11110000
+
+
+There are few ways to achieve this:
+
+- Using ``debug_level`` directive in domain section in ``sssd.conf``.
+
+.. code-block:: bash
```
Comment (pbrezina) [inline src/troubleshooting/backend.rst:123]: ini

File: src/troubleshooting/backend.rst [hunk 10]
```diff
@@ -1,12 +1,626 @@
...
+    ...
+    [domain/IPA.PB]
+    ...
+    debug_level = 0x3ff0
+    ...
+
+- Using ``sss_debuglevel`` tool which will set the level only for current SSSD
+  instance and will set the original level when SSSD is restarted.
+
+.. code-block:: bash
```
Comment (pbrezina) [inline src/troubleshooting/backend.rst:135]: console

File: src/troubleshooting/backend.rst [hunk 11]
```diff
@@ -1,12 +1,626 @@
...
+
+    # systemctl start sssd.service
+    # sss_debuglevel 0x3ff0
+
+- Using ``sssctl`` util:
+
+.. code-block:: bash
```
Comment (pbrezina) [inline src/troubleshooting/backend.rst:142]: console

**Review by pbrezina — CHANGES_REQUESTED** [review]
Minor comments - set the correct linter.

* `text` - plain text
* `sssd-log` - sssd logs
* `bash` - bash script
* `console` - terminal (with command prompt PS1 or PS2)
* `ini` - sssd.conf

Please distinguish between bash script and terminal. Bash script is just bash commands, terminal (console) also contains the command prompt. We probably will extend the syntax highlighting to make to look better so it is important to distinguish between those two. Use bash where console is not really important.

File: src/troubleshooting/backend.rst (hunk 1 above)
Comment (elkoniu) [reply src/troubleshooting/backend.rst]: Can you please take a look if in the latest version it is more clear?

File: src/troubleshooting/backend.rst (hunk 7 above)
Comment (elkoniu) [reply src/troubleshooting/backend.rst]: Fixed

File: src/troubleshooting/backend.rst (hunk 9 above)
Comment (elkoniu) [reply src/troubleshooting/backend.rst]: Fixed

File: src/troubleshooting/backend.rst (hunk 11 above)
Comment (elkoniu) [reply src/troubleshooting/backend.rst]: Fixed

File: src/troubleshooting/backend.rst (hunk 8 above)
Comment (elkoniu) [reply src/troubleshooting/backend.rst]: Fixed

File: src/troubleshooting/backend.rst (hunk 1 above)
Comment (pbrezina) [reply src/troubleshooting/backend.rst]: Ack.

Comment (pbrezina) [reply src/troubleshooting/backend.rst]: @justin-stephenson Do you agree with the change as well?

Comment (justin-stephenson) [reply src/troubleshooting/backend.rst]: s/Cache is a local database/The SSSD Cache is a local database/

Otherwise Ack from me

Comment (elkoniu) [reply src/troubleshooting/backend.rst]: Fixed

**Comment by justin-stephenson** [comment]
Ack from my side.

**Pull request merged**